src/v2/Apps/Partner/Components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,27 @@
+import React from "react"
+import { Media } from "v2/Utils/Responsive"
+import { Swiper } from "@artsy/palette"
+import { DesktopCarousel } from "./DesktopCarousel"
+
+export interface CarouselProps {
+  children: (itemsPerViewport: number) => JSX.Element | JSX.Element[]
```

Comment:
This makes me nervous about how difficult this is going to be to upgrade to V3 when the time comes. Ideally we wouldn't have any of this custom handling.

Reply:
Added another branch with the refactoring of the carousel. You can find it here: https://github.com/artsy/force/commit/1ea7fae5b3b19a637dd384bd155687aa692e5947

But I,m not sure that it solves the problem

Reply:
OK so I synced up with Jonathan: this is fine for now. Whichever branch you're happy with works for me. It seems that the V3 variation of this will just use it's default behavior and we won't need to port this use-case over to that so we can just simply replace it when the time comes.

Reply:
I'll create a PR for the branch above after this one will be merged.